apps/frontend/core-dapp/src/features/trade/TradeStore.ts

```diff
@@ -576,7 +585,29 @@ export class TradeStore {
 
     runInAction(() => {
       this.closingPosition = false
-      if (!error) this.closePositionValue = ''
+      if (!error) this.userClosePositionValue = ''
+    })
+  }
+
+  private async redeemPosition(): Promise<void> {
+    const { closePositionAmountBN, selectedMarket } = this
+
+    if (closePositionAmountBN === undefined || selectedMarket === undefined) return
+
+    this.closingPosition = true
+    const { error } = await selectedMarket.redeem(
+      this.direction === 'long' ? closePositionAmountBN : BigNumber.from(0),
+      this.direction === 'short' ? closePositionAmountBN : BigNumber.from(0)
+    )
```

Comment:
If market is resolved user should just redeem whatever they have remaining in that market. So it should just redeem with user's long/short balance I reckon, remember to add a `balance === undefined` check above

```suggestion
    const { error } = await selectedMarket.redeem(
      this.direction === 'long' ?  this.selectedMarket.longTokenBalanceBN : BigNumber.from(0),
      this.direction === 'short' ? this.selectedMarket.shortTokenBalanceBN : BigNumber.from(0)
    )
```

Reply:
oh it does seem we also have to pass a recipient as a third parameter?

Reply:
my bad, the third parameter is only when we migrate to new interfaces, could you add a TODO comment above to add signer's address as recipient later when we switch to new interfaces?

Reply:
In this case `closePositionAmountBN` equals `longTokenBalanceBN`/`shortTokenBalanceBN`, but I agree that your way is clearer.